### DISCUSSION TITLE
Intercepting Prisma with msw or nock

### DISCUSSION THREAD

**Discussion by EvHaus** [prisma/prisma#12640] [Q&A]
I'm hoping to be able to intercept all network requests that Prisma makes using [msw](https://mswjs.io/docs/getting-started/integrate/node) (or maybe a similar tool like `nock`), but it's not working by default. My goal is to do this for the purpose of writing unit tests. I read through the official [unit testing docs](https://www.prisma.io/docs/guides/testing/unit-testing) and it seems to be relying on mocking prisma itself, as opposed to mocking the network requests. I'm curious, what does Prisma use under-the-hood to make its network requests?

`msw` intercepts [these native modules](https://github.com/mswjs/interceptors#what-this-library-does) so I guess Prisma is not using any of those?

**Update:** I guess Prisma uses `undici` which [doesn't support this](https://github.com/nodejs/undici/issues/1002) flow :(

**Comment by janpio** [comment]
Which network requests are you trying to intercept? The calls to the databases, or when Prisma downloads the engine files?

**Reply by EvHaus** [reply]
The calls to the database.

**Reply by janpio** [reply]
Those are not made by Node, but by the Prisma Engines (which are executables or native node libraries) under the hood. Any Node ecosystem tools won't work there unfortunately. In specific modes we use Undici to talk to an internal HTTP server that could potentialyl be intercepted, but the "normal" operation does not - and you most probably want to test as close to what you would actually run in your application of course - making this a bad idea.

**Reply by EvHaus** [reply]
Yeah, I understand. I ended up relying on integration tests which test against a real Prisma database to get around it, but was hoping to be able to setup light-weight unit tests that can test isolated parts of my business logic without having to worry about a real database being in place too.

Thanks for the answer!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
